### ISSUE TITLE
🐛 Bug Report: Upgrade to 1.11.0 broke msgraph provider

### ISSUE THREAD

**Issue by jwnmulder** [backstage/backstage#16361]
### 📜 Description

It seems that #16000 broke the ingestion of Azure AD users/groups.

This change results in the wrong scope being used.

```diff
    const token = await this.tokenCredential.getToken(
-      'https://graph.microsoft.com/.default',
+      `${this.baseUrl}/.default`,
```

In version 1.10.0 the scope is `https://graph.microsoft.com/.default`. In version 1.11.0 the scope is `https://graph.microsoft.com/v1.0/.default` which is not valid.

### 👍 Expected behavior

msgraph provider being able to ingest users and groups correctly

### 👎 Actual Behavior with Screenshots

na

### 👟 Reproduction steps

Upgrade Backstage from 1.10.0 to 1.11.0

### 📃 Provide the context for the Bug.

From the logs

catalog error MicrosoftGraphOrgEntityProvider:default refresh failed invalid_resource: 500011 - [2023-02-15 14:18:10Z]: AADSTS500011: The resource principal named https://graph.microsoft.com/v1.0 was not found in the tenant named .... This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.

### 🖥️ Your Environment

OS:   Linux 5.15.79.1-microsoft-standard-WSL2 - linux/x64
node: v16.19.0
yarn: 1.22.10
cli:  0.22.2 (installed)
backstage:  1.11.0

Dependencies:
  @backstage/app-defaults                               1.2.0
  @backstage/backend-app-api                            0.4.0
  @backstage/backend-common                             0.18.2
  @backstage/backend-dev-utils                          0.1.0
  @backstage/backend-plugin-api                         0.4.0
  @backstage/backend-tasks                              0.4.3
  @backstage/catalog-client                             1.3.1
  @backstage/catalog-model                              1.2.0
  @backstage/cli-common                                 0.1.11
  @backstage/cli                                        0.22.2
  @backstage/config-loader                              1.1.8
  @backstage/config                                     1.0.6
  @backstage/core-app-api                               1.5.0
  @backstage/core-components                            0.11.2, 0.12.4
  @backstage/core-plugin-api                            1.4.0
  @backstage/errors                                     1.1.4
  @backstage/eslint-plugin                              0.1.0
  @backstage/integration-aws-node                       0.1.1
  @backstage/integration-react                          1.1.10
  @backstage/integration                                1.4.2
  @backstage/plugin-adr-backend                         0.3.0
  @backstage/plugin-adr-common                          0.2.6
  @backstage/plugin-adr                                 0.4.0
  @backstage/plugin-allure                              0.1.31
  @backstage/plugin-api-docs                            0.9.0
  @backstage/plugin-app-backend                         0.3.42
  @backstage/plugin-auth-backend                        0.18.0
  @backstage/plugin-auth-node                           0.2.11
  @backstage/plugin-badges-backend                      0.1.36
  @backstage/plugin-badges                              0.2.39
  @backstage/plugin-catalog-backend-module-gitlab       0.1.13
  @backstage/plugin-catalog-backend-module-ldap         0.5.9
  @backstage/plugin-catalog-backend-module-msgraph      0.5.0
  @backstage/plugin-catalog-backend-module-openapi      0.1.8
  @backstage/plugin-catalog-backend                     1.7.2
  @backstage/plugin-catalog-common                      1.0.11
  @backstage/plugin-catalog-graph                       0.2.27
  @backstage/plugin-catalog-graphql                     0.3.18
  @backstage/plugin-catalog-import                      0.9.5
  @backstage/plugin-catalog-node                        1.3.3
  @backstage/plugin-catalog-react                       1.3.0
  @backstage/plugin-catalog                             1.8.0
  @backstage/plugin-explore-common                      0.0.1
  @backstage/plugin-explore-react                       0.0.26
  @backstage/plugin-explore                             0.4.0
  @backstage/plugin-github-actions                      0.5.15
  @backstage/plugin-graphiql                            0.2.47
  @backstage/plugin-graphql-backend                     0.1.32
  @backstage/plugin-home                                0.4.31
  @backstage/plugin-jenkins-backend                     0.1.32
  @backstage/plugin-jenkins-common                      0.1.13
  @backstage/plugin-jenkins                             0.7.14
  @backstage/plugin-org                                 0.6.5
  @backstage/plugin-permission-common                   0.7.3
  @backstage/plugin-permission-node                     0.7.5
  @backstage/plugin-permission-react                    0.4.10
  @backstage/plugin-proxy-backend                       0.2.36
  @backstage/plugin-scaffolder-backend                  1.11.0
  @backstage/plugin-scaffolder-common                   1.2.5
  @backstage/plugin-scaffolder-node                     0.1.0
  @backstage/plugin-scaffolder-react                    1.1.0
  @backstage/plugin-scaffolder                          1.11.0
  @backstage/plugin-search-backend-module-elasticsearch 1.1.3
  @backstage/plugin-search-backend-module-pg            0.5.3
  @backstage/plugin-search-backend-node                 1.1.3
  @backstage/plugin-search-backend                      1.2.3
  @backstage/plugin-search-common                       1.2.1
  @backstage/plugin-search-react                        1.5.0
  @backstage/plugin-search                              1.1.0
  @backstage/plugin-shortcuts                           0.3.7
  @backstage/plugin-sonarqube-backend                   0.1.7
  @backstage/plugin-sonarqube-react                     0.1.2
  @backstage/plugin-sonarqube                           0.6.3
  @backstage/plugin-tech-insights-backend-module-jsonfc 0.1.26
  @backstage/plugin-tech-insights-backend               0.5.8
  @backstage/plugin-tech-insights-common                0.2.10
  @backstage/plugin-tech-insights-node                  0.4.0
  @backstage/plugin-tech-insights                       0.3.7
  @backstage/plugin-tech-radar                          0.6.1
  @backstage/plugin-techdocs-backend                    1.5.3
  @backstage/plugin-techdocs-module-addons-contrib      1.0.10
  @backstage/plugin-techdocs-node                       1.5.0
  @backstage/plugin-techdocs-react                      1.1.3
  @backstage/plugin-techdocs                            1.5.0
  @backstage/plugin-user-settings-backend               0.1.6
  @backstage/plugin-user-settings                       0.7.0
  @backstage/release-manifests                          0.0.8
  @backstage/test-utils                                 1.2.5
  @backstage/theme                                      0.2.17
  @backstage/types                                      1.0.2
  @backstage/version-bridge                             1.0.3
Done in 0.54s.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by TaylorRadke** [comment]
I can verify that I am also having the same issue after bumping to 1.11.0

**Comment by drodil** [comment]
Sorry for this, my mistake! Apparently, this was not verified further before the release and I didn't have time to check it in the pre-release version either. I have pushed a revert for this for now and maybe make another config value for this in the future.

**Comment by QuadmanSWE** [comment]
For our deployment, we downgraded "@backstage/plugin-catalog-backend-module-msgraph to version "0.4.7" and managed to get our integration to msgraph to work again. Using that until this is merged and readily available.

**Comment by afscrome** [comment]
@drodil  I don't think you'll need a separate config option for the token scope, you should be able to derive the scope from the hostname of the target- `https://${target.hostname}/.default`

https://learn.microsoft.com/en-us/graph/sdks/national-clouds?tabs=csharp
https://learn.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints

**Comment by benjdlambert** [comment]
@afscrome @drodil so I merged the rollback for now, will get that shipped today, if you want to put a PR together for the actual fix that's fine and we can get it in next release.

**Comment by benjdlambert** [comment]
Closing as got released part of the bonus 1.11.1 release.
